### PR TITLE
Provider/aws Apply security group after restoring db_instance from snapshot

### PR DESCRIPTION
**Pull request by m-s-austin**
Currently awsgosdk (and terraform by extension) assign a newly restored snapshot only the default security group. This patch causes the resource to check if the vpc_security_group_ids list is set, and if it is it waits for the restored DB to become available and then updates it in order to get the settings in the right state.

### CODE REVIEW

File: builtin/providers/aws/resource_aws_db_instance.go [hunk 1]
```diff
@@ -350,6 +350,42 @@ func resourceAwsDbInstanceCreate(d *schema.ResourceData, meta interface{}) error
 		if err != nil {
 			return fmt.Errorf("Error creating DB Instance: %s", err)
 		}
+
+		if attr := d.Get("vpc_security_group_ids").(*schema.Set); attr.Len() > 0 {
+			var s []*string
+			for _, v := range attr.List() {
+				s = append(s, aws.String(v.(string)))
+			}
+			log.Printf("[INFO] DB is restoring from snapshot with default security, but should be set to %s", v.(string))
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_db_instance.go]: `v.(string)` here should be `s`, the `[]*string` slice

**Comment by catsby** [comment]
One nitpick, otherwise looks good.

Curious, did you consider doing an update db API call directly, instead of passing off to the update method?

**Comment by m-s-austin** [comment]
I did consider it, and felt it would be unnecessarily duplicating code, plus I didn't see any immediate negative to just using the update method. Did you see some reason to prefer one way or the other?

File: builtin/providers/aws/resource_aws_db_instance.go (hunk 1 above)
Comment (m-s-austin) [inline builtin/providers/aws/resource_aws_db_instance.go]: good catch, thanks I will fix shortly

**Comment by catsby** [comment]
My concern would be any untended or otherwise unexpected updates being applied

**Comment by m-s-austin** [comment]
I’d think it would be no different than calling terraform apply again, so it should properly track which resources are dirty or not?

**Comment by catsby** [comment]
Thats probably OK then, thanks!

**Pull request merged**